Flush API update in mobile SDK (#598)
rudderlabs/rudderstack-docs: docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-react-native-sdk.mdx

rudderClient.putAnonymousId(ANONYMOUS_ID)
```

## `flush` API

> React Native SDK supports the `flush()` API. It retrieves all the messages present in the database, divides them into individual batches and flushes them to the Rudder Server.

Reviewer: ```suggestion
The React Native SDK supports the `flush()` API. It retrieves all the messages present in the database, divides them into individual batches and flushes them to the RudderStack backend.
```

> For example, if the `flushQueueSize` is 30 and there are 180 events in the database advanced the `flush()` API is called, the React Native SDK will retrieve all those events and divide them into batches of 30 messages each, that is, total 6 batches. 

Reviewer: ```suggestion
For example, if the `flushQueueSize` is 30 and there are 180 events in the database when the `flush()` API is called, the SDK will retrieve all the events and divide them into batches of 30 messages each, that is, in 6 batches. 
```

If a batch fails for some reason, the remaining batches are dropped as well to maintain the sequence of the messages. A batch is considered as failed if it isn’t sent to the server even after 3 retries. 

> In the device mode, the `flush()` API also calls the destination SDK’s `flush()` API (if exists).  

Reviewer: ```suggestion
In the device mode, the `flush()` API also calls the destination SDK’s `flush()` API (if applicable).  
```
